pkg/agent/agent.go

```diff
@@ -0,0 +1,129 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package agent provides the main entrypoint for the agent.
+package agent
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"strings"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/siderolabs/siderolink/pkg/wireguard"
+)
+
+// Config is the configuration for the agent.
+//
+//nolint:govet
+type Config struct {
+	WireguardEndpoint string   // WireguardEndpoint is the endpoint for the Wireguard server.
+	APIEndpoint       string   // APIEndpoint is the gRPC endpoint for the SideroLink API.
+	JoinToken         string   // JoinToken is the join token for the SideroLink API.
+	ForceUserspace    bool     // ForceUserspace forces the usage of the userspace UDP device for Wireguard.
```

Comment:
i think moving this as the last element should fix govet, at least this is a simple struct which seems easily fixable

Reply:
Thanks! Fixed that as well as several other relatively simple cases.